### ISSUE TITLE
Pa11y crashes with puppeteer error

### ISSUE THREAD

**Issue by josebolos** [pa11y/pa11y#487]
From: https://github.com/pa11y/pa11y-dashboard/issues/242

Trying to run the latest version of pa11y (5.2.0) on `http://culturaestero.regione.emilia-romagna.it/it` makes it crash:

```
pa11y http://culturaestero.regione.emilia-romagna.it/it

Welcome to Pa11y

 > Running Pa11y on URL http://culturaestero.regione.emilia-romagna.it/it

Error: Evaluation failed: TypeError: Cannot read property 'process' of undefined
    at __puppeteer_evaluation_script__:73:19
    at new Promise (<anonymous>)
    at runHtmlCodeSniffer (__puppeteer_evaluation_script__:72:11)
    at runPa11y (__puppeteer_evaluation_script__:34:32)
    at ExecutionContext._evaluateInternal (/Users/jose.bolos/.nvm/versions/node/v8.16.1/lib/node_modules/pa11y/node_modules/puppeteer/lib/ExecutionContext.js:122:13)
    at <anonymous>
    at process._tickCallback (internal/process/next_tick.js:189:7)
  -- ASYNC --
    at ExecutionContext.<anonymous> (/Users/jose.bolos/.nvm/versions/node/v8.16.1/lib/node_modules/pa11y/node_modules/puppeteer/lib/helper.js:111:15)
    at DOMWorld.evaluate (/Users/jose.bolos/.nvm/versions/node/v8.16.1/lib/node_modules/pa11y/node_modules/puppeteer/lib/DOMWorld.js:112:20)
    at <anonymous>
    at process._tickCallback (internal/process/next_tick.js:189:7)
  -- ASYNC --
    at Frame.<anonymous> (/Users/jose.bolos/.nvm/versions/node/v8.16.1/lib/node_modules/pa11y/node_modules/puppeteer/lib/helper.js:111:15)
    at Page.evaluate (/Users/jose.bolos/.nvm/versions/node/v8.16.1/lib/node_modules/pa11y/node_modules/puppeteer/lib/Page.js:833:43)
    at Page.<anonymous> (/Users/jose.bolos/.nvm/versions/node/v8.16.1/lib/node_modules/pa11y/node_modules/puppeteer/lib/helper.js:112:23)
    at runPa11yTest (/Users/jose.bolos/.nvm/versions/node/v8.16.1/lib/node_modules/pa11y/lib/pa11y.js:220:29)
    at <anonymous>
    at process._tickCallback (internal/process/next_tick.js:189:7)
```

**Comment by jsa34** [comment]
Noticed same issue occurring when running against https://www.bbc.co.uk/

**Comment by steveburrows81** [comment]
We're getting the same error when running Pa11y directly or via the dashboard on https://www.dundee.ac.uk. 

Some pages work fine, eg: https://www.dundee.ac.uk/undergraduate/dentistry

**Comment by josebolos** [comment]
I've reverted the change that was causing this behaviour and released it as [pa11y 5.2.1](https://github.com/pa11y/pa11y/releases/tag/5.2.1).

Please let us know if it doesn't fix your problem.

**Comment by steveburrows81** [comment]
That's working now, thank you for the quick fix.

**Comment by joeyciechanowicz** [comment]
This is now completely fixed in Pa11y `5.3.0` which adds support for sites using AMD.
